integrations/extensions/starter-kits/search-with-milvus/search-with-watsonx-data-milvus.md

```diff
@@ -0,0 +1,117 @@
+
+### Option 2: Ingest data using custom code
+Here is a sample code to ingest documents into Milvus: [../search-with-custom-service/examples/index-with-milvus.py](../search-with-custom-service/examples/index-with-milvus.py). To run it, 
+1. Install dependencies:
+   ```bash
+   python3 -m pip install pymilvus langchain langchain-milvus langchain-ibm ibm-watsonx-ai PyPDF2
+   ```
+2. Create environment variables for Milvus credentials
+  ```bash
+  export MILVUS_HOST="Your Milvus GRPC host"
+  export MILVUS_PORT="Your Milvus GRPC port"
+  export MILVUS_USER="ibmlhapikey" // The default username for watsonx.data Milvus
+  export MILVUS_PASSWORD="Your watsonx.data API key"
+
+  export MILVUS_COLLECTION_NAME="Your Milvus collection name" // It can be anything
+  export WATSONX_AI_APIKEY="Your watsonx.ai API key" // watsonx.ai embeddings model is used to create vectors
+  export WATSONX_AI_PROJECT_ID="Your watsonx.ai project ID" // watsonx.ai project ID is required to access the embeddings models
+  ```
+3. Update the `SOURCE_FILES`, `SOURCE_URLS`, and `SOURCE_TITLES` variables at the begining of the script to your file names, urls, and titles respectively.
+4. Run the script
+   ```bash
+   python3 index-with-milvus.py
+   ```
+
+## Step 3: Connect to watsonx Orchestrate or Assistant
+There are two ways to connect your Milvus vector database to watsonx Orchestrate or Assistant for search and conversational search, either through the built-in Milvus search integration or using the Custom Service search.
+
+**NOTE: The embeddings model(s) you use for search using either option in [Step 3](#step-3-connect-to-watsonx-orchestrate-or-assistant) has to align with the embeddings model(s) used for data ingestion in [Step 2](#step-2-ingest-data-into-milvus).**
+
+### Option 1: Use the built-in Milvus search integration
+This option allows you to integrate with your watsonx.data Milvus service through a built-in feature of watsonx Orchestrate/Assistant.
+
+Please go to the [watsonx Assistant product docs](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-search-overview) to find more details about how to set up the built-in Milvus search integration.
+
+### Option 2: Use the Custom Service search
+Custom Service search provides a more advanced and flexible way to connect your Milvus vector database to watsonx Orchestrate or Assistant. It requires you to set up a custom server that must be reachable by your watsonx Orchestrate or Assistant instance. Then you need to configure the custom server by providing your server credentials and metadata.
+
+With this option, you can implement more advanced search capabilities with Milvus, such as,
+* Use any embedding models: https://milvus.io/docs/embeddings.md
```

Comment:
Suggest:  Flexibility to use any of the embedding models supported by Milvus, and not only those provided by watsonx.ai.  For a list of embeddings supported by Milvus see: https://milvus.io/docs/embeddings.md